include/hxString.h

```diff
@@ -155,6 +155,7 @@ class HXCPP_EXTERN_CLASS_ATTRIBUTES String
    inline const char *&raw_ref() { return __s; }
    inline const char *raw_ptr() const { return __s; }
    const char *utf8_str(hx::IStringAlloc *inBuffer = 0,bool throwInvalid=true) const;
+   const char *utf8_str_len(hx::IStringAlloc *inBuffer = 0,int *byteLength = 0,bool throwInvalid=true) const;

```

Comment:
I'm terrible at C++: is overloading `utf8_str` an option here?

Reply:
Wasn't sure if it would be better to add another optional argument or to make another function. Thinking about it some more I think the extra argument is better, so `utf8_str_len` should be gone now.